src/store/index.js

```diff
@@ -1,22 +1,38 @@
 import api from '@/api/index'
+import uuid from 'uuid/v4'
 
 export default {
   strict: true,
 
   state: {
-    loading: true,
+    loading: false,
+    newTask: {
+      description: '',
+      estimatedSize: 5
+    },
     tasks: {
       edges: [],
       pageInfo: {}
     }
   },
 
   actions: {
-    async TASKS ({ commit }, variables) {
+    async TASKS ({ state, commit }, variables) {
+      if (state.loading) { return }
       commit('START_LOADING')
       const payload = await api.Tasks(variables)
       commit('UPDATE_TASKS', payload)
       commit('FINISH_LOADING')
+    },
+
+    async CREATE_TASK ({ state, commit }) {
+      if (state.loading) { return }
+      commit('START_LOADING')
```

Comment:
I'm just curious, Is there any reason to split process to lots of commit?

Reply:
I can not naming mutation a (START_LOADING + MAKE_MUTATION_ID_TASK) and mutation b (PUSH_TASK + UPDATE_NEW_TASK + FINISH_LOADING).